### DISCUSSION TITLE
How to use Default Case in Union Topic ? (C++)

### DISCUSSION THREAD

**Discussion by rla0h** [OpenDDS/OpenDDS#4782] [Q&A]
Hello, I learned Union Topic Type in OpenDDS, but I had a problem in learning.
I use this Union Topic IDL.

module Test {
    enum TypeKind {
        STRING_TYPE,
        LONG_TYPE,
        FLOAT_TYPE
    };

    @topic
    union UnionTypeTest switch(@key TypeKind) {
    case STRING_TYPE:
        string string_data;
    case LONG_TYPE:
        long long_data;
    case FLOAT_TYPE:
        float float_data;
    default:
        boolean default_data;    
    };
  };

I successed to print STRING_TYPE, LONG_TYPE, FLOAT_TYPE data, but I can't data about default. 
Based on the debugging, it appears that the serialization is failing.
How could I see default_data in Union Topic Type in C++?

Thank you.

**Comment by rla0h** [comment]
sorry, This is my mistake..
I fixed it, using Union Topic

**Reply by jwillemsen** [reply]
The IDL you have is illegal, the default case isn't allowed to be there, I created a tao_idl issue, see https://github.com/DOCGroup/ACE_TAO/issues/2279
